### ISSUE TITLE
Unable to get trialEndsAt date without payment up front after trial expired

### ISSUE THREAD

**Issue by RobertBoes** [laravel/cashier-paddle#85]
- Cashier Paddle Version: 1.2.2
- Laravel Version: 8.10.0
- PHP Version: 7.4.7
- Database Driver & Version: MySQL 5.7.29

### Description:
After a trial has expired (without payment up front) the `trialEndsAt` method tries to get the trial_ends_at date from the subscription instead of the customer.

### Steps To Reproduce:
Create a user, then use the following to simulate a trial without payment up front that is expired:
```
$user->createAsCustomer([
    'trial_ends_at' => now()->subDays(10)
]);
```
When calling `$user->trialEndsAt()` the following exception is thrown: https://flareapp.io/share/J7oEp0Vm#F73

**Comment by yoeriboven** [comment]
You'd probably want it to always return the value of `trial_ends_at` on the `customers` table if it is set.

There's one edge case however. What if you also (at a later stage) set a trial on the subscription model. In that case it should return the newest value. This seems _very_ uncommon though. 

I'll implement the first option later. If the maintainers want the second option, let me know. :)

**Comment by RobertBoes** [comment]
Yes, the user is no longer `onGenericTrial`, because the trial has ended, that makes sense.

I think `onGenericTrial` should not be used to get the trial_ends_at value. Maybe the method should become something like this:
```php
public function trialEndsAt($name = 'default')
{
    if (is_null($this->subscription($name))) {
        return $this->customer->trial_ends_at;
    }

    return $this->subscription($name)->trial_ends_at;
}
```

This way the trial_ends_at of the customer is returned when the user has no subscription. But, this probably has other implications, because the customer might not exist.

**Comment by yoeriboven** [comment]
I don't think that would be a problem.

If you don't have either a customer or a subscription you shouldn't call this method anyway.

I proposed a fix with #87. Thanks for creating the issue.
